bld/CLMBuildNamelist.pm

```diff
@@ -2856,39 +2627,28 @@ sub setup_logic_hydrology_switches {
   if ( $h2osfcflag == 1 && $subgrid != 1 ) {
     $log->fatal_error("if h2osfcflag is ON, subgridflag can NOT be off!");
   }
-  # These should NOT be set for CLM5.0 and beyond
```

Comment:
It's not clear to me if this should be removed. I thought of these as variables we'd be able to remove at some point. But, they might be useful for NWP. So perhaps this should just be removed. The if could be changed, to just allow it for clm4_5 (since the numeric representation was removed).

Reply:
I felt it would be a maintenance headache to hard-code specific physics versions these options are and are not allowed for. As you say, they might be useful for NWP, so we'd have to have an ever-growing list of specific versions these are and are not allowed for. We can possibly revisit this when we have a more robust way to get the general physics option ("clm", "nwp", etc.) and the version of that option. However, my opinion on this is that it's unnecessary work on our part, and it's good enough to document these in namelist_definition as being deprecated.